keel-sqs/src/main/kotlin/com/netflix/spinnaker/keel/sqs/SqsResourceCheckListener.kt

```diff
@@ -27,38 +34,48 @@ internal class SqsResourceCheckListener(
   private val objectMapper: ObjectMapper,
   private val actuator: ResourceActuator
 ) {
-  private lateinit var listenerJob: Job
+  private lateinit var rootJob: Job
 
-  private val scope: CoroutineScope
-    get() = CoroutineScope(Default)
+  private val scope: CoroutineScope = CoroutineScope(IO)
 
   @PostConstruct
   fun startListening() {
-    listenerJob = scope.launch {
-      while (isActive) {
-        handleMessages()
+    rootJob = scope.launch {
+      val channel = Channel<Message>()
+      launchMessageReceiver(channel)
+      repeat(10) {
+        launchWorker(channel)
       }
     }
   }
 
-  internal fun handleMessages() {
-    with(sqsClient) {
-      receiveMessages()
+  private fun CoroutineScope.launchMessageReceiver(channel: SendChannel<Message>) = launch {
+    runUntilCancelled {
+      sqsClient
+        .receiveMessages()
         .also {
           if (it.isNotEmpty()) log.debug("Got {} messages: {}", it.size, it.map(Message::getMessageId))
         }
         .forEach {
-          try {
-            with(it.parse()) {
-              actuator.checkResource(name.let(::ResourceName), apiVersion, kind)
-            }
-            deleteMessage(it)
-          } catch (e: JsonProcessingException) {
-            log.error("Could not parse message payload: {}", it.body)
-          } catch (e: Throwable) {
-            log.error("Resource Actuator threw {}: {}", e.javaClass.simpleName, e.message)
+          channel.send(it)
```

Comment:
is there a version of send that will fail if it can't enqueue the message within some timeframe? would probably be better than blocking indefinitely here, because we could just reset the visibilityTimeout to zero and let another instance receive the message if so

Reply:
This is a coroutine channel send, not an SQS operation.

Reply:
I don't see any override of this method, unfortunately.